.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseMergedDateTimeExtractor.cs

```diff
@@ -242,6 +242,16 @@ private void AddTo(List<ExtractResult> dst, List<ExtractResult> src, string text
                         }
                         else
                         {
+                            if (result.Metadata?.IsHoliday ?? false)
```

Comment:
Whenever some new nested condition like this is introduced, please add a code comment describing the rationale and examples of the situations it's designed to cover.